app/routes/new-signup.js

```diff
@@ -0,0 +1,22 @@
+import Route from '@ember/routing/route';
+import ENV from 'website-my/config/environment';
+
+export default class NewSignupRoute extends Route {
+  async model() {
+    const response = await fetch(`${ENV.BASE_API_URL}/users/self`, {
+      credentials: 'include',
+    });
+    const userData = await response.json();
+    if (response.status === 401) {
+      alert('You have not logged in. Please login first to fill this form.');
+      window.open(
+        'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97',
```

Comment:
Should this URL be coming from a constants file?

Reply:
This same URL is being used in all the routes to check if the users are logged in or not, do not want to change all the places right now. Will create a separate ticket and raise a PR separately for this.